### ISSUE TITLE
NativeScript app consuming a lib that imports another lib

### ISSUE THREAD

**Issue by sryzycki** [sryzycki/nx-tns#2]
@NathanWalker I'm sorry to bother you but I've another question to run by you! I've created a branch to isolate the case: https://github.com/sryzycki/nx-tns/tree/lib-consuming-a-lib

When you develop using Nx Workspaces there will be a time when a lib (that is consumed by a NativeScript app) needs to import another lib (let's say it's just a few lines of vanilla TypeScript code). Currently when you run `tns run ios --emulator --syncAllFiles` it would throw the error below because we haven't run a `npm i ../../libs/utils`.

It feels like it would be a big overhead if one needed to install dependencies of dependencies inside of a NativeScript app's package.json.

```
JavaScript error:
file:///app/tns_modules/@nx-tns/common-nx-tns/src/appender.pipe.js:6:24: JS ERROR Error: Could not find module '@nx-tns/utils'.
Computed path '/Users/samuel/Library/Developer/CoreSimulator/Devices/4AD8EB7C-0DDD-4385-B6B0-4083DE86AC21/data/Containers/Bundle/Application/32F0E1CD-74BF-4AE6-B5B3-3D5BF435D031/mobile.app/app/tns_modules/@nx-tns/utils'.
```

What do you think? Is there a cure for this at the moment or am I doing something completely wrong?

**Comment by NathanWalker** [comment]
Yes there is a cure for this. You want to devise a good strategy around your lib structure to handle the essentials to avoid ‘barrel hell’ if you will. Jeff Whelpley and I will be presenting a number of things at ngConf in April covering this and other topics. In meantime, you really want a best practice set of libs as your main barrels, ie:
* api (if you’re using shared api models, etc that may be autogeneated by a backend)
* core (low level Singleton services and State like auth, network, locale, etc)
* features (this may grow to be most expansive)
* scss (shared variables and other styling across web and mobile)
* utils 

Out of the above: `core` and `utils` are completely portable and can be interchangeable across many different workspaces. To some extent so can `scss`.

`api` and `features` are usually specific to your particular multiplatform project.

That gives you 5 barrels to install into your {N} project ever. 

Our talk at ngConf will explain this better and unveil some extra niceness as well.

Sorry for delayed response, very busy leading up to the conference.

**Comment by sryzycki** [comment]
@NathanWalker I'm happy that you managed to find some time to reply at all! Thanks a lot for writing this up.

This granular Nx Workspace approach is pretty new to us in my current company so I think I'll wait for you guys to shed some light on this Nx + NativeScript during NgConf.

It would be awesome to see an example of a Nx Workspace with multiple web apps AND multiple mobile apps that share let's say 80% code between them - as this is our challenge at the moment.

I'll stay tuned, thanks!
